internal/query/cache.go

```diff
@@ -30,6 +32,21 @@ func startCaches(background context.Context, connectors connector.Connectors, in
 	if err != nil {
 		return nil, err
 	}
+	caches.authnKeys, err = connector.StartCache[
+		AuthnKeyIndex,
+		string,
+		*CachedPublicKey](
+		background,
+		[]AuthnKeyIndex{
+			AuthnKeyIndexKeyIDAndIdentifier,
+		},
+		cache.PurposeAuthNKeys,
+		connectors.Config.Instance,
```

Comment:
Please create add an `AuthNKey` field in the `CachesConfig` to allow custom configuration of this cache.